src/machine/machine_stm32_rng.go

```diff
@@ -18,6 +18,12 @@ func GetRNG() (uint32, error) {
 	cnt := RNG_MAX_READ_RETRIES
 	for !stm32.RNG.SR.HasBits(stm32.RNG_SR_DRDY) {
 		cnt--
+		if stm32.RNG.SR.HasBits(stm32.RNG_SR_CECS) {
```

Comment:
Should these two checks be before the loop?

Reply:
Reference manual :

CECS => "When the clock error detection is enabled (CED = 0) and if the RNG clock frequency is too
low, the RNG sets to “1” both the CEIS and CECS bits to indicate that a clock error
occurred."  

SECS => "When a noise source (or seed) error occurs, the RNG stops generating random numbers
and sets to “1” both SEIS and SECS bits to indicate that a seed error occurred. If a value is
available in the RNG_DR register, it must not be used as it may not have enough entropy."

... So it seems to be even better to do the checks before the loop. 

Thanks for the suggestion.